apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -66,6 +67,32 @@ private ServerAction(TimeSpan preInterval, IEnumerable<string> msgs, RpcExceptio
             public int? DeliveryAttempt { get; }
         }
 
+        // Class to simulate exceptions in Acknowledgement and/or ModifyAcknowledgementDeadline calls.
+        private class AckModifyAckDeadlineAction
+        {
+            public static AckModifyAckDeadlineAction Data(Exception ex, int numberOfFailures, bool onAck, bool onModifyAckDeadline) => new AckModifyAckDeadlineAction(ex, numberOfFailures, onAck, onModifyAckDeadline);
+
+            public static AckModifyAckDeadlineAction BadAck(Exception ex, int numberOfFailures) => new AckModifyAckDeadlineAction(ex, numberOfFailures, true, false);
+            
+            public static AckModifyAckDeadlineAction BadModifyAckDeadline(Exception ex, int numberOfFailures) => new AckModifyAckDeadlineAction(ex, numberOfFailures, false, true);
+            
+            private AckModifyAckDeadlineAction(Exception exception, int numberOfFailures, bool onAck, bool onModifyAckDeadline)
+            {
+                Exception = exception;
+                NumberOfFailures = numberOfFailures;
+                OnAck = onAck;
+                OnModifyAckDeadline = onModifyAckDeadline;
+            }
+
+            public Exception Exception { get; }
+
+            public int NumberOfFailures { get; }
+
+            public bool OnAck { get; }
```

Comment:
I don't understand what this or `OnModifyAckDeadline` actually mean. I'd suggest either renaming, or adding an XML summary comment. (If you add a comment, then ignore the bit above about removing blank lines.)

Reply:
Reading a bit more, I see what it's doing, but I suspect there may be a cleaner design. We can revisit once this is in though.

Reply:
> I don't understand what this or `OnModifyAckDeadline` actually mean. I'd suggest either renaming, or adding an XML summary comment. (If you add a comment, then ignore the bit above about removing blank lines.)
Added the XML comments.

> Reading a bit more, I see what it's doing, but I suspect there may be a cleaner design. We can revisit once this is in though.
Sure Jon.